src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -4503,64 +4503,74 @@ class MavlinkStreamDistanceSensor : public MavlinkStream
 
 	unsigned get_size() override
 	{
-		return _distance_sensor_sub->is_published() ? (MAVLINK_MSG_ID_DISTANCE_SENSOR_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES) : 0;
+		return MAVLINK_MSG_ID_DISTANCE_SENSOR_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES;
 	}
 
 private:
-	MavlinkOrbSubscription *_distance_sensor_sub;
-	uint64_t _dist_sensor_time;
+	MavlinkOrbSubscription *_distance_sensor_sub[ORB_MULTI_MAX_INSTANCES] {};
```

Comment:
Ah, yeah.  This is a little simpler than my suggestion of using templates for all the instances you want.

Reply:
Yeah, I just did the same as `MavlinkStreamBatteryStatus` does. Tested on a hardware and I seem to be getting data fine.